tycho-indexer/src/extractor/dynamic_contract_indexer.rs

```diff
@@ -0,0 +1,348 @@
+    tracer: Arc<dyn EntryPointTracer<Error = ()>>,
+    protocol_entrypoints: HashMap<ComponentId, Vec<EntryPointWithData>>,
+    entrypoint_results: HashMap<EntryPointWithData, TracedEntryPoint>,
+}
+
+// TODO: Things to check:
+// If I emit an AccountDelta of type Update without code or balance, will it overwrite the existing
+// value?
+impl DynamicContractIndexer<()> {
+    pub fn new(
+        protocol: String,
+        entrypoint_gw: Arc<dyn EntryPointGateway>,
+        storage_source: Arc<dyn AccountExtractor<Error = ()>>,
+        tracer: Arc<dyn EntryPointTracer<Error = ()>>,
+    ) -> Self {
+        Self {
+            protocol,
+            entrypoint_gw,
+            storage_source,
+            tracer,
+            protocol_entrypoints: HashMap::new(),
+            entrypoint_results: HashMap::new(),
+        }
+    }
+
+    /// This function will initialize the DynamicContractIndexer
+    /// It will load from the gateway all the entrypoints + entrypoint for the protocols and the
+    /// respective trace results.
+    pub async fn initialize(&mut self) {
+        let entrypoint_filter = EntryPointFilter::new(Some(self.protocol.clone()));
+
+        let entrypoints = self
+            .entrypoint_gw
+            .get_entry_points_with_data(entrypoint_filter)
+            .await
+            .unwrap();
+
+        for entrypoint in entrypoints {
+            self.protocol_entrypoints
+                .insert(entrypoint.clone());
+
+            let traced_entrypoints = self
+                .entrypoint_gw
+                .get_traced_entry_point(entrypoint)
+                .await
+                .unwrap();
+            if let Some(traced_entrypoint) = traced_entrypoints {
+                self.entrypoint_results
+                    .insert(entrypoint, traced_entrypoint);
+            }
+        }
+    }
+
+    pub async fn process_block_update(
+        &mut self,
+        new_entrypoints: &HashMap<ComponentId, Vec<EntryPointWithData>>,
+        changes: &Vec<TransactionAccountUpdates>,
+        block: &Block,
+    ) -> DCIBlockUpdate {
+        let mut result_deltas: HashMap<Transaction, Vec<AccountDelta>> = HashMap::new();
+        let mut result_components: HashMap<ComponentId, Vec<Address>> = HashMap::new();
+
+        // Registers new Entrypoints sent by the extractor to the map of Component -> Entrypoints
+        for (component_id, entrypoints) in new_entrypoints {
+            for entrypoint in entrypoints {
+                self.protocol_entrypoints
+                    .get(&component_id)
+                    .get_or_insert_default()
+                    .push(entrypoint.clone());
+            }
+        }
+
+        // Select for analysis the newly detected EntryPointsWithData that haven't been analyzed
+        // yet. This filter prevents us from re-analyzing entrypoints that have already been
+        // analyzed, which can be a case if all the components have the same entrypoint.
```

Comment:
Hmm it's not clear to me why this is necessary? In this first iteration I assumed that we will:

1. Get new entrypoint
2. Analyze
3. Finish

So not sure why we have two collections? Are we already implementing some kind of retry mechanism here?